pkg/cache/v3/linear.go

```diff
@@ -314,90 +425,86 @@ func (cache *LinearCache) GetResources() map[string]types.Resource {
 	return resources
 }
 
-func (cache *LinearCache) CreateWatch(request *Request, _ Subscription, value chan Response) (func(), error) {
+func (cache *LinearCache) CreateWatch(request *Request, sub Subscription, value chan Response) (func(), error) {
 	if request.GetTypeUrl() != cache.typeURL {
 		return nil, fmt.Errorf("request type %s does not match cache type %s", request.GetTypeUrl(), cache.typeURL)
 	}
-	// If the version is not up to date, check whether any requested resource has
-	// been updated between the last version and the current version. This avoids the problem
-	// of sending empty updates whenever an irrelevant resource changes.
-	stale := false
-	staleResources := []string{} // empty means all
-
-	// strip version prefix if it is present
-	var lastVersion uint64
-	var err error
-	if strings.HasPrefix(request.GetVersionInfo(), cache.versionPrefix) {
-		lastVersion, err = strconv.ParseUint(request.GetVersionInfo()[len(cache.versionPrefix):], 0, 64)
-	} else {
-		err = errors.New("mis-matched version prefix")
+
+	// If the request does not include a version the client considers it has no current state.
+	// In this case we will always reply to allow proper initialization of dependencies in the client.
+	ignoreCurrentSubscriptionResources := request.GetVersionInfo() == ""
+	if !strings.HasPrefix(request.GetVersionInfo(), cache.versionPrefix) {
```

Comment:
I'm wondering if this logic cannot be abstracted away in the `ResponseWatch`. I think the `ignoreCurrentSubscriptionResources` is functionally irrelevant. What seems to matter is the "real" set of returned resources.